tests/unittest/_torch/auto_deploy/unit/singlegpu/test_ad_speculative_decoding.py

```diff
@@ -0,0 +1,75 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from _model_test_utils import get_small_model_config
+from build_and_run_ad import ExperimentConfig, main
+
+from tensorrt_llm.llmapi import DraftTargetDecodingConfig, KvCacheConfig
+
+
+def test_ad_speculative_decoding_smoke():
+    """Test speculative decoding with AutoDeploy using the build_and_run_ad main()."""
+
+    # Use a simple test prompt
+    test_prompt = "What is the capital of France?"
+
+    # Get base model config
+    experiment_config = get_small_model_config("meta-llama/Meta-Llama-3.1-8B-Instruct")
+    speculative_model_dir = get_small_model_config("TinyLlama/TinyLlama-1.1B-Chat-v1.0")["args"][
+        "model"
+    ]
+
+    print(f"Speculative model path: {speculative_model_dir}")
+    # Configure speculative decoding with a draft model
+    spec_config = DraftTargetDecodingConfig(
+        max_draft_len=3, speculative_model_dir=speculative_model_dir
+    )
+
+    # Configure KV cache
+    kv_cache_config = KvCacheConfig(
+        free_gpu_memory_fraction=0.1,
```

Comment:
can it be smaller than that? the config returned by `get_small_model_config` has free_mem_ratio --> we can re-use the same value